python/lsst/faro/base/CatalogMeasurementBase.py

```diff
@@ -192,3 +195,28 @@ def _getReferenceCatalog(self, butlerQC, dataIds, refCats, filterList, epoch=Non
         refCatFrame = hstack([refCatTable, refCat.asAstropy()]).to_pandas()
 
         return refCatFrame
+
+    def _persistMeasurementInputs(self, config, shelveName, **kwargs):
+        """Persist in-memory objects sent as inputs to metric measurement run method.
+
+        This function is intended to be used for development and testing purposes
+        as a debug tool and is NOT to be used in routine operation.
+
+        Parameters
+        ----------
+        config : `lsst.pex.config.Config`
+            Config to be saved.
+        shelveName : `str`
+            Filename for output shelve.
+        """
+
+        import shelve
+        shelf = shelve.open(shelveName, 'n')
+
+        shelf['config'] = config
+        for key in kwargs.keys():
+            try:
+                shelf[key] = kwargs[key]
+            except TypeError:
+                print('ERROR shelving: {0}'.format(key))
+        shelf.close()
```

Comment:
`shelve.open()` can be used as a context manager so please switch to that rather than the explicit close().

Reply:
Thanks. Following this suggestion.